packages/cadl-ranch-specs/http/models/dictionary/main.cadl

```diff
@@ -0,0 +1,53 @@
+import "@azure-tools/cadl-azure-core";
+import "@azure-tools/cadl-ranch-expect";
+
+using Cadl.Http;
+using Azure.Core.Foundations;
+
+@serviceTitle("Dictionary in Complex Test Service")
+@serviceVersion("2016-02-29")
+@server("http://localhost:3000", "Testserver endpoint")
+@doc("Illustrates dictionary generation, as well as serialization and deserialization.")
+namespace Cadl.Testserver.BodyComplexDictionary;
+
+model dictionaryWrapper {
+  defaultProgram?: Record<string>;
+}
+
+@scenario
+@route("/complex/dictionary/typed")
+namespace dictionary {
+  @route("/valid")
+  @scenarioDoc("Put complex types with dictionary properties")
+  @put
+  op putValid(
+    @doc("Please put a dictionary with 5 key-value pairs: \"txt\":\"notepad\", \"bmp\":\"mspaint\", \"xls\":\"excel\", \"exe\":\"\", \"\":null")
```

Comment:
The `@doc` property is more for describing the property, i.e. what we would generate if this was a client library. The description of what the mock api tests should expect should be in the `@scenarioDoc` header. See the docs for more info https://github.com/Azure/cadl-ranch/blob/main/docs/writing-scenario-spec.md